Add the ability to specify zoom level

captainmedic opened the pull request:
These changes add onto the already existing ability to append to ?lat={{lat}}&lon={{lon}} a &zoom={{zoom}}

## Description
Changes were made to the files that are associated with the ability to call map specifics from the address bar of the browser.

## Motivation and Context
I thought about adding this pull request for those who utilize third party alarm systems such as pokealarm. Pokealarm has the ability to link users straight to google maps at GPS coordinates, however, having the ability to channel users to the map creators website could have a couple of benefits. One the website owner would be able to track hits from the pokealarm for statistical purposes. It would also allow for the most accurate time remaining and any additional info about Pokestops in the area and even other Pokemon in the nearby vicinity.

## How Has This Been Tested?
This was tested on a Windows testing environment, to ensure that it would not run into any errors.

## Screenshots (if appropriate):
![example](https://cloud.githubusercontent.com/assets/27046411/26212000/be5f2f2a-3bb9-11e7-8d7a-033d38ecd08a.png)


## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--  NOTE: In order to check code style locally and avoid having your build rejected by Travis, -->
<!--  run the following commands before you commit: `flake8 .` and `npm run lint`. Fix any -->
<!--  issues they point out. Note also that flake's NOQA is disabled on Travis. -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

captainmedic commented:
I do welcome comments from more experienced coders on the implementation of this. I am sure it is botchy as my python skills are very little. Also to have it successfully work in the map.js file I changed `zoom:Store.get('zoomLevel')` to `zoom: centerZoom` the variable that is now assigned in the map.html. With this change it appears that if you resize the page and refresh it goes back to the default zoom specified which is 16. Would there be a better way to implement this?

peterchabs commented:
Awesome!
I've been wanting this feature for a bit, hope it works out!

friscoMad commented:
There is no need to pass the value from falsk to html, just parse it from the UI.
It is more code but cleaner, you could also use the same for lat and long.
See: http://stackoverflow.com/questions/901115/how-can-i-get-query-string-values-in-javascript

Also that way you can check if the user has some zoom stored and use it when it is not url defined, as right now you are removing the feature of storing the zoom between visits.

captainmedic commented:
Thanks for the suggestion. I have implemented this method of calling the resource instead of  making the changes to app.py. I have removed those commits and pushed the other method. Currently it is working well on my windows test build. Now user saved zoom levels will remain in effect unless specified by `&zoom=`

friscoMad commented on static/js/map.js:
```diff
@@ -114,6 +114,10 @@ function removePokemonMarker(encounterId) { // eslint-disable-line no-unused-var
 }
 
 function initMap() { // eslint-disable-line no-unused-vars
+    if (centerZoom) { // checks to see if specified in url
```
You shouldn't store it, just use it when centering the map.
Otherwise we are overriding his stored zoom when opening a zoomed map.

friscoMad commented on templates/map.html:
```diff
@@ -420,8 +420,18 @@
     <script>
-      var centerLat = {{lat}};
-      var centerLng = {{lng}};
+        function getParameterByName(name, url) {
```
Add the function directly to map.js
Drop lat/long parsing in app.py as we will be getting it from js

captainmedic replied:
Thanks again for the suggestions. I have moved the function itself to map.js. However, when in the map.js file we lose functionality of it, in that it will no longer use the zoom that is define by url. I have tried several different methods to no avail, at least on my windows environment. I will try again tomorrow when I am not so tired to figure out why it doesn't work when moving the function, if you have any ideas I'd be open to it.

sebastienvercammen commented on templates/map.html:
```diff
@@ -420,8 +420,18 @@
+            if (!url) url = window.location.href;
+            name = name.replace(/[\[\]]/g, "\\$&");
+            var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+                results = regex.exec(url);
```
Oh dios, mis ojos. Take a look at `window.location.search`.

friscoMad commented:
@sebastienvercammen this is ready to be merged.
